src/screens/EmailForm/index.tsx

```diff
@@ -43,25 +43,25 @@ const EmailForm = ({ route, navigation }: EmailFormProps) => {
   }, []);
 
   useEffect(() => {
-    setInvalidEmail(!email.match(EMAIL_REGEX));
-    if (!email.match(EMAIL_REGEX) && isValidationAttempted) setErrorMessage('Votre e-mail n\'est pas valide.');
-    else setErrorMessage('');
+    const isEmailInvalid = !email.match(EMAIL_REGEX);
+    dispatch(isEmailInvalid
+      ? { type: SET_ERROR, payload: isValidationAttempted ? 'Votre e-mail n\'est pas valide' : '' }
+      : { type: RESET_ERROR });
```

Comment:
je trouve que ce n'est pas tres claire de mettre la condition `isEmailInvalid` dans le `dispatch`
et pareil pour la condition `isEmailInvalid`


qu'est ce qui t'a fait pousser a ne pas faire comme suit : 
```
if (!email.match(EMAIL_REGEX) && isValidationAttempted) dispatch({ type: SET_ERROR, payload: 'Votre e-mail n\'est pas valide.' });
else dispatch({ type: RESET_ERROR });
```
Intuitivement c'est ce que j'aurais fait...
C'est une vraie question

Reply:
j'ai homogénéisé avec ce qu'on a dit plus haut. DONE